src/name.cc

```diff
@@ -6,35 +6,44 @@
 
 #include <algorithm>
 #include <cstring>
+#include <cctype>
 
 // name - Naming Table
 // http://www.microsoft.com/typography/otspec/name.htm
 
 namespace {
 
-bool ValidInPsName(char c) {
-  return (c > 0x20 && c < 0x7f && !std::strchr("[](){}<>/%", c));
+// We disallow characters outside the URI spec "unreserved characters"
+// set; any chars outside this set will be replaced by underscore.
+bool AllowedInPsName(char c) {
+  return isalnum(c) || std::strchr("-._~", c);
 }
 
-bool CheckPsNameAscii(const std::string& name) {
+bool SanitizePsNameAscii(std::string& name) {
+  if (name.size() > 63)
+    return false;
+
   for (unsigned i = 0; i < name.size(); ++i) {
-    if (!ValidInPsName(name[i])) {
-      return false;
+    if (!AllowedInPsName(name[i])) {
+      name[i] = '_';
```

Comment:
Just in case it is a completely invalid name (I  have seen these), we should return `false` if all characters were replaced by underscores.

Reply:
It wouldn't really do any harm if the name ended up as "_______", would it?

Reply:
Of course not, it just looks strange.